idaes/core/unit_model.py

```diff
@@ -92,6 +92,20 @@ def build(self):
         """
         super(UnitModelBlockData, self).build()
 
+        # Add a placeholder for referecnes to costing blocks
+        self._costing_block_ref = None
+
+        # Check for overloading of initialize method
+        # TODO: Remove in IDAES v2.0
+        if type(self).initialize != UnitModelBlockData.initialize:
```

Comment:
Performance note: we re-test this for *every* instance of *every* unit model.  This could add up.  I am not sure if I know of a better approach, though.

Reply:
There aren't usually a lot of unit models.  I doubt it would have any significant impact.